### PR TITLE
Add tree-invocation-order lazy engine to test build ordering.

### PR DESCRIPTION
**Pull request by whastings**
This adds a dummy engine for testing this change to ember-engines: https://github.com/ember-engines/ember-engines/pull/289

### CODE REVIEW

File: lib/tree-invocation-order/addon/templates/application.hbs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lazy/addon/templates/application.hbs
```
Comment (nathanhammond) [inline lib/tree-invocation-order/addon/templates/application.hbs]:
```diff
- lazy
+ tree-invocation-order
```

File: lib/tree-invocation-order/package.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  "keywords": [
+    "ember-addon",
+    "ember-engine"
+  ],
+  "dependencies": {
+    "ember-ajax": "*",
+    "ember-cli-babel": "^5.1.6",
+    "ember-cli-htmlbars": "*",
+    "ember-engines": "*"
+  }
```
Comment (nathanhammond) [inline lib/tree-invocation-order/package.json:13]:
```diff
- ember-engines
- ember-ajax
+ broccoli-funnel
+ broccoli-merge-trees
+ broccoli-stew
```

File: lib/tree-invocation-order/addon/lazy-import-target.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export default {};
```
Comment (nathanhammond) [inline lib/tree-invocation-order/addon/lazy-import-target.js]: Rename: `tree-invocation-order-import-target.js`

File: lib/tree-invocation-order/addon/routes.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import buildRoutes from 'ember-engines/routes';
+import importTarget from './lazy-import-target';
```
Comment (nathanhammond) [inline lib/tree-invocation-order/addon/routes.js]: `import importTarget from './tree-invocation-order-import-target' ;`

File: lib/tree-invocation-order/addon/routes/lazy-route.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export default {};
```
Comment (nathanhammond) [inline lib/tree-invocation-order/addon/routes/lazy-route.js]: Rename: `tree-invocation-order-route.js`

**Review by nathanhammond — CHANGES_REQUESTED** [review]
Looks good, a few tiny fixes.

**Comment by whastings** [comment]
Thanks! I've made the fixes!

**Review by nathanhammond — APPROVED** [review]

**Pull request merged**